multiversx/mx-chain-go
# Fix out of gas save key value

## Reasoning behind the pull request
- SaveKeyValue built-in function can, on some occasions, return a negative (overflowed) GasRemaining value. The SC processor correctly handles this but the function should take care of the gas consumption logic
  
## Proposed changes
- added a final check in the SaveKeyValue built-in function

## Testing procedure
- standard system test
- scenario:
    * in epoch 1 call the `SaveKeyValue` built-in function with enough gas. Save a random key with a random value.  
    example:
   ```
   SaveKeyValue@aa@bb
   ```
    * before epoch 3, execute the scenario, trying to send the same transaction data but with less than the expected gas limit (say, put the gas limit to the minimum value to make the transaction execute - eg. only the tx data extra gas). Log ERROR prints will appear and the transaction will fail with the error: `Gas consumed exceeded`
    * after the epoch 3, re-do the scenario. No more log error prints will occur and the transaction will fail with a not-enough-gas error.
    
## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
